subworkflows/local/generate_downstream_samplesheets/main.nf

```diff
@@ -0,0 +1,120 @@
+                def fastq_2             = !meta.single_end ? out_path + reads[1].getName() : ""
+                def fasta               = ""
+            [ sample: sample, run_accession:run_accession, instrument_platform:instrument_platform, fastq_1:fastq_1, fastq_2:fastq_2, fasta:fasta ]
+        }
+
+    channelToSamplesheet(ch_list_for_samplesheet,"${params.outdir}/downstream_samplesheets/taxprofiler", format)
+
+}
+
+workflow SAMPLESHEET_MAG {
+    //
+    // MAG doesn't take PE-data & SE-data in a single samplesheet
+    //
+    take:
+    ch_reads
+
+    main:
+    format = 'csv' // most common format in nf-core
+
+    // Combine the short and long reads belonging to the same sample
+    ch_reads
+        .map{meta, reads ->
+                tuple( groupKey(meta.id - "_longReads", 2), meta, reads)
+            }
+        .groupTuple(remainder: true)
+        .map{key, meta, reads ->
+                new_meta = [
+                    id: key,
+                    run: key,
+                    single_end: meta[0].single_end,
+                    long_reads: meta[0]?.long_reads ?: meta[1]?.long_reads ?: false
+                    ]
+                // Making sure the long reads are the final element of the array.
+                read_files = reads.flatten().sort(false){ a, b -> a.getName().tokenize('.')[0] <=> b.getName().tokenize('.')[0] }
+            [new_meta, read_files]
+            }
+        .tap{ ch_reads_grouped }
+
+    // Make your samplesheet channel construct here depending on your downstream
+    ch_list_for_samplesheet = ch_reads_grouped
+        .map {
+            meta, reads ->
+                def out_path       = file(params.outdir).toString() + '/filter/filtered/'
+                def sample         = meta.id
+                def run            = meta.run
+                def group          = ""                                                                                       // only used for co-abundance in binning
+                def short_reads_1  = meta.long_reads == (reads.size() > 2) ? out_path + reads[0].getName() : ""               // If long reads, but no short reads, then short_reads_1 is empty
+                def short_reads_2  = meta.long_reads == (reads.size() > 2) && reads[1] ? out_path + reads[1].getName() : ""
+                def long_reads     = meta.long_reads ? out_path + reads.last().getName() : ""                                  // If long reads, take final element
+            [sample: sample, run: run, group: group, short_reads_1: short_reads_1, short_reads_2: short_reads_2, long_reads: long_reads]
+        }
+        .branch{
+            se: it.short_reads_2 ==""
+            pe: true
+        }
+
+    // Throw a warning that only long reads are not supported yet by MAG
+    ch_list_for_samplesheet
+        .pe
+        .filter{ it.long_reads !="" && it.short_reads_1=="" }
```

Comment:
>  Didn't think people will go into the output.md docs to check for that disclaimer.

Sorry to clarify: you can still have the warning, but we should also actually remove those lines _for_ the user from the samplesheet (in addition to the warning log, and the output.md description).

I would make it as easier for users to just take the samplesheet and run, rather than having to edit.

I suspect users are also not likley to read the docs, but it means they will just give the samplesheet straight away to mag 'blindly', and then mag will fail with those long-read only lines.

We are looking for an as streamlined experience as possible so removing entirely will go towards that.

Reply:
Makes sense. lets do it like that.